apps/gp2-frontend/src/auth/Signin.tsx

```diff
@@ -36,7 +36,12 @@ const Signin: React.FC<Record<string, never>> = () => {
       <UtilityBar>
         <WelcomePage
           onClick={signin}
-          authFailed={searchParams.has('state') && searchParams.has('error')}
+          authFailed={
+            (searchParams.has('state') &&
+              searchParams.has('error') &&
+              'invalid') ||
+            undefined
```

Comment:
Is this the same as this ternary

```suggestion
            (searchParams.has('state') &&
              searchParams.has('error')) ? 'invalid' :  undefined
```

Reply:
yeah i think so and i prefer ternary too, i just thought in FE we wanted to go all FP and use these instead of ternaries